src/Handler/StackageHome.hs

```diff
@@ -53,7 +53,8 @@ instance ToJSON SnapshotInfo where
 getStackageDiffR :: SnapName -> SnapName -> Handler TypedContent
 getStackageDiffR name1 name2 = track "Handler.StackageHome.getStackageDiffR" $ do
     cacheSeconds $ 60 * 60 * 48
-    Entity sid1 _ <- lookupSnapshot name1 >>= maybe notFound return
+    Entity sid1 prevSnap <- lookupSnapshot name1 >>= maybe notFound return
+    prevprevSnapName <- fromMaybe name1 . map snd <$> snapshotBefore (snapshotName prevSnap)
```

Comment:
Wouldn't it be better to keep this as a `Maybe`? `Nothing` is a much better sentinel value for indicating that the previous-previous snapshot doesn't exist.

```suggestion
    prevprevSnapName <- map snd <$> snapshotBefore (snapshotName prevSnap)
```

Reply:
I agree, good catch :+1:  (in fact I had thought of this but then forgot to do that part this morning)